File: HISTORY - CODE/Comatose v1.2.1 [UNRELEASED]/C++/human.cpp
#include "human.h"
#include "lvlchange.h"


human::human(whole locSt, arrow dirSt)
{
	locOFhh = locSt;
	locOFh = &locOFhh;
	dirOFhh = dirSt;
	dirOFh = &dirOFhh;
	dead = false;
}


human::~human()
{
}

/* hstraight
* - moves human
*/
void human::hstraight(CMAX w)
{
	switch (*dirOFh)
	{
	case arrow::U:
		*locOFh -= w;
		break;
	case arrow::D:
		*locOFh += w;
		break;
	case arrow::L:
		*locOFh -= 1;
		break;
	case arrow::R:
		*locOFh += 1;
		break;
    case arrow::BAD:
        break; //WARNING
	}
}

/* printh
* - moves human
* - returns false if loses
* - prints location to board
*/
bool human::printh(SYMVECT& board, CMAX w)
{
	hstraight(w);
	bool dead = false;
	//if ( board.size() < static_cast<whole>(*locOFh) ) return true;
	dead = board[*locOFh] != symbol::E && board[*locOFh] != symbol::P;
	if (!dead) board[*locOFh] = symbol::F;
	return !dead;
}

/* hturn
* - waits for input
* - if E then plants a bomb
* - else changes direction of human
*/
void human::hturn(CNT& bombs, bomb& B)
{
	char c = '\0';
	c = _getch();
	if (!B.plantbomb(c, *locOFh))
	{
		arrow check = symbols::ar(c);
		if (arrow::BAD != check) *dirOFh = check;
	}
	if ('f' == c) lvlchange::chfast();
}

/* nextlvll
* - returns true if move to next level
*/
bool human::nextlvll(MAX crack, MAX crachv)
{
	if (locOFhh == crack || locOFhh == crack + crachv || locOFhh == crack - crachv)
	{
		if (creature::opening())
		{
			return true;
		}
	}
	return false;
}

File: src/Python2 [CONVERTED]/pyComa/human.py
from __future__ import absolute_import
from symbol import symbol, arrow, symbols, getch

import creature
import bomb
import pickup
import sys
import lvlchange
import board

class human(object):
    def __init__(self, locSt, dirSt):
        self.loc_ = locSt
        self.dir_ = dirSt
        self.dead = False
    def hstraight(self):
        u''' hstraight
        * - moves human
        '''
        for comp, result in [(arrow.U, -board.WIDTH),
                            (arrow.D, +board.WIDTH),
                            (arrow.L, -1),
                            (arrow.R, +1)
                            ]:
            if comp == self.dir():
                self.loc_ += result
                break
        if not self.dir():
            raise ValueError
    def printh(self):
        u''' printh
        * - moves human
        * - returns false if loses
        * - prints location to board
        '''
        self.hstraight()
        dead = False
        #if ( board.size() < static_cast<whole>(*loc_) ) return true;
        dead = board.BOARD_REF.Boarder(self.loc_) != symbol.E and board.BOARD_REF.Boarder(self.loc_) != symbol.P
        if not dead:
            board.BOARD_REF.Boarder(self.loc_, symbol.F)
        return not dead
    def hturn(self, B):
        u''' hturn
        * - waits for input
        * - if E then plants a bomb
        * - else changes direction of human
        '''
        c = u''
        c = getch()
        if not B.plantbomb(c, self.loc_):
            check = symbols.ar(c)
            if check != arrow.BAD: #should be != 'BAD'
                self.dir(check)
        if not self.dir():
            raise ValueError
        if c == u'f':
            lvlchange.chfast()
    def nextlvll(self, crack, crachv):
        u''' nextlvll
        * - returns true if move to next level
        '''
        if self.loc_ == crack or self.loc_ == crack + crachv or self.loc_ == crack - crachv:
            if creature.opening:
                return True
        return False
    def loc(self):
        return self.loc_
    def dir(self, new=None):
        if not new is None:
            self.dir_ = new
        return self.dir_
